src/test/java/TraineeServiceTest.java

```diff
@@ -0,0 +1,95 @@
+import org.example.dao.TraineeDao;
+import org.example.models.Trainee;
+import org.example.service.TrainingService;
+import org.example.service.impl.TraineeServiceImpl;
+import org.example.util.Generator;
+import org.example.util.UserUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class TraineeServiceTest {
+
+    @Mock
+    private TraineeDao traineeDao;
+
+    @Mock
+    private UserUtil userUtil;
+    @Mock
+    private TrainingService trainingService;
+
+    @InjectMocks
+    private TraineeServiceImpl traineeService;
+
+    @BeforeEach
+    public void setup() {
+        traineeService.setUserUtil(userUtil);
+    }
+
+    @Test
+    public void createTraineeTest() {
+        Trainee trainee = new Trainee();
+        trainee.setFirstName("John");
+        trainee.setLastName("Doe");
+
+        List<String> existingUsernames = new ArrayList<>();
+        when(userUtil.getAllExistingUsernames()).thenReturn(existingUsernames);
+
+        try (MockedStatic<Generator> generatorMockedStatic = mockStatic(Generator.class)) {
+            generatorMockedStatic.when(() -> Generator.generateUsername("John", "Doe", existingUsernames))
+                    .thenReturn("John.Doe");
+            generatorMockedStatic.when(Generator::generatePassword)
+                    .thenReturn("password123");
+            generatorMockedStatic.when(() -> Generator.generateId(anyList()))
+                    .thenReturn(1L);
+
+
+            traineeService.createTrainee(trainee);
+
+            verify(userUtil, times(1)).getAllExistingUsernames();
+            verify(traineeDao, times(1)).create(trainee);
+            assertEquals("John.Doe", trainee.getUsername());
```

Comment:
Nit: Usually, it is better to use AsserJ lib for assertions instead of jUnit
```suggestion
            assertThat( trainee.getUsername().isEqual("John.Doe");
```